src/components/Button.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+  <button :class="type" @click="onClick">
```

Comment:
IMHO, we shouldn't pass the _type_ property to the class without check if it is a valid button type. What do you think about check if the type exists and if not render the default button?